plugins/GSdx/GSUtil.cpp

```diff
@@ -396,6 +396,34 @@ D3D_FEATURE_LEVEL GSUtil::CheckDirect3D11Level(IDXGIAdapter *adapter, D3D_DRIVER
 	return SUCCEEDED(hr) ? level : (D3D_FEATURE_LEVEL)0;
 }
 
+GSRendererType GSUtil::GetBestRenderer()
+{
+	CComPtr<IDXGIAdapter1> adapter;
+	DXGI_ADAPTER_DESC1 desc;
+	CComPtr<IDXGIFactory1> dxgi_factory;
+
+	if (!GSUtil::CheckDXGI() || !GSUtil::CheckD3D11())
+		return GSRendererType::DX9_HW;
```

Comment:
> https://msdn.microsoft.com/en-gb/library/windows/desktop/bb776913(v=vs.85).aspx

Is that a joke? Maybe adding a macro for ` if (SUCCEEDED(hr)) {` would be nice. Then everything could be nicely written as paragraph without line breaks. Strange what microsoft offers as solution.... I really recommend using code refactoring tools... It will show you if you nest too deep.